src/httpserver.cpp

```diff
@@ -389,12 +389,21 @@ static void HTTPWorkQueueRun(WorkQueue<HTTPClosure> *queue)
 }
 
 /** libevent event log callback */
-static void libevent_log_cb(int severity, const char *msg)
+static void libevent_log_cb(int severity, const char *_msg)
 {
 #ifndef EVENT_LOG_WARN
 // EVENT_LOG_WARN was added in 2.0.19; but before then _EVENT_LOG_WARN existed.
 #define EVENT_LOG_WARN _EVENT_LOG_WARN
 #endif
+    std::string msg(_msg);
+    if (msg.empty())
+        return;
+
+    // messages to this call back sometimes seem to have an
+    // extra newline that can be removed.
```

Comment:
do we have an idea why this is happening?

Reply:
Nope, I have no clue. Yes, this is a workaround for an unknown cause. I didn't check, but maybe only some of the message lines are impacted, which would mean that they have a logging function that itself adds a newline which is passed a newline terminated string. But that is just a guess.